src/dma.rs

```diff
@@ -30,51 +32,68 @@ pub enum Half {
     Second,
 }
 
+pub trait CharacterMatch {
+    /// Checks to see if the peripheral has detected a character match and
+    /// clears the flag
+    fn check_character_match(&mut self, clear: bool) -> bool;
+}
+
 /// Frame reader "worker", access and handling of frame reads is made through this structure.
-pub struct FrameReader<BUFFER, CHANNEL, const N: usize>
+pub struct FrameReader<BUFFER, PAYLOAD, const N: usize>
 where
     BUFFER: Sized + StableDeref<Target = DMAFrame<N>> + DerefMut + 'static,
 {
     buffer: BUFFER,
-    channel: CHANNEL,
+    payload: PAYLOAD,
     matching_character: u8,
 }
 
-impl<BUFFER, CHANNEL, const N: usize> FrameReader<BUFFER, CHANNEL, N>
+impl<BUFFER, PAYLOAD, const N: usize> FrameReader<BUFFER, PAYLOAD, N>
 where
     BUFFER: Sized + StableDeref<Target = DMAFrame<N>> + DerefMut + 'static,
 {
     pub(crate) fn new(
         buffer: BUFFER,
-        channel: CHANNEL,
+        payload: PAYLOAD,
         matching_character: u8,
-    ) -> FrameReader<BUFFER, CHANNEL, N> {
+    ) -> FrameReader<BUFFER, PAYLOAD, N> {
         Self {
             buffer,
-            channel,
+            payload,
             matching_character,
         }
     }
 }
 
+impl<BUFFER, PAYLOAD, CHANNEL, const N: usize> FrameReader<BUFFER, RxDma<PAYLOAD, CHANNEL>, N>
```

Comment:
Hmm, I suppose it would also be possible to just implement the trait here

Reply:
I think this is the right place :)